retrofit/src/main/java/retrofit2/Retrofit.java

```diff
@@ -158,12 +159,14 @@ private void eagerlyValidateMethods(Class<?> service) {
     }
   }
 
-  ServiceMethod loadServiceMethod(Method method) {
-    ServiceMethod result;
+  ServiceMethod<?, ?> loadServiceMethod(Method method) {
+    ServiceMethod<?, ?> result = serviceMethodCache.get(method);
+    if (result != null) return result;
+
     synchronized (serviceMethodCache) {
```

Comment:
Oops that wasn't supposed to be in this PR but a separate one. Hard to do anything better since we don't have Java 8's `computeIfAbsent` to override.